transformers/transformers_simple.py

```diff
@@ -0,0 +1,95 @@
+"""
+Optuna example for fine-tuning a BERT-based text classification model on the IMDb dataset
+with hyperparameter optimization using Optuna. In this example, we fine-tune a lightweight
+pre-trained BERT model on a small subset of the IMDb dataset to classify movie reviews as
+positive or negative. We optimize the validation accuracy by tuning the learning rate
+and batch size.To learn more, you can check the following
+documentation: https://huggingface.co/docs/transformers/en/hpo_train?
+"""
+
+from datasets import load_dataset
+import evaluate
+import torch
+
+from transformers import AutoModelForSequenceClassification
+from transformers import AutoTokenizer
+from transformers import set_seed
+from transformers import Trainer
+from transformers import TrainingArguments
+
+
+set_seed(42)
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+train_dataset = load_dataset("imdb", split="train").shuffle(seed=42).select(range(1000))
+valid_dataset = load_dataset("imdb", split="test").shuffle(seed=42).select(range(500))
+
+model_name = "prajjwal1/bert-tiny"
+tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+
+def tokenize(batch):
+    return tokenizer(batch["text"], padding="max_length", truncation=True, max_length=512)
+
+
+tokenized_train = train_dataset.map(tokenize, batched=True).select_columns(["input_ids", "label"])
+tokenized_valid = valid_dataset.map(tokenize, batched=True).select_columns(["input_ids", "label"])
```

Comment:
Sorry, my previous suggestion was wrong. Clearly attention mask should be here.
```suggestion
tokenized_train = train_dataset.map(tokenize, batched=True).select_columns(["input_ids", "attention_mask", "label"])
tokenized_valid = valid_dataset.map(tokenize, batched=True).select_columns(["input_ids", "attention_mask", "label"])
```